Tracking issue for CI build failure "Installing openssl@3 has failed!"

Several CoreCLR and Mono CI lanes failed with an error `Not found PathtoPublish: /Users/runner/work/1/s/artifacts/log/`.

PRs that encountered the issue (not a full list):
 - https://github.com/dotnet/runtime/pull/80426
 - https://github.com/dotnet/runtime/pull/80362
 - https://github.com/dotnet/runtime/pull/80423
 - https://github.com/dotnet/runtime/pull/80136

## Error Message
<!-- Fill for repository issues. For infrastructure issues the engineering services (@dotnet/dnceng) is going to fill it. -->
```json 
{
  "ErrorMessage":"Installing openssl@3 has failed!",
  "BuildRetry": false
} 
```
<!--Known issue error report start -->
### Report
|Build|Definition|Step Name|Console log|Pull Request|
|---|---|---|---|---|
|[132055](https://dev.azure.com/dnceng-public/public/_build/results?buildId=132055)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/132055/logs/38)|dotnet/runtime#80435|
|[132019](https://dev.azure.com/dnceng-public/public/_build/results?buildId=132019)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/132019/logs/74)|dotnet/runtime#80433|
|[131641](https://dev.azure.com/dnceng-public/public/_build/results?buildId=131641)|dotnet/runtime|Install native dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/131641/logs/1319)|dotnet/runtime#80392|
|[131642](https://dev.azure.com/dnceng-public/public/_build/results?buildId=131642)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/131642/logs/93)|dotnet/runtime#80392|
|[131643](https://dev.azure.com/dnceng-public/public/_build/results?buildId=131643)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/131643/logs/46)|dotnet/runtime#80392|
|[132043](https://dev.azure.com/dnceng-public/public/_build/results?buildId=132043)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/132043/logs/40)|dotnet/runtime#80164|
|[132033](https://dev.azure.com/dnceng-public/public/_build/results?buildId=132033)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/132033/logs/64)|dotnet/runtime#80421|
|[132040](https://dev.azure.com/dnceng-public/public/_build/results?buildId=132040)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/132040/logs/14)|dotnet/runtime#80434|
|[132039](https://dev.azure.com/dnceng-public/public/_build/results?buildId=132039)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/132039/logs/30)|dotnet/runtime#80434|
|[132010](https://dev.azure.com/dnceng-public/public/_build/results?buildId=132010)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/132010/logs/35)|dotnet/runtime#80432|
|[132009](https://dev.azure.com/dnceng-public/public/_build/results?buildId=132009)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/132009/logs/56)|dotnet/runtime#80432|
|[132002](https://dev.azure.com/dnceng-public/public/_build/results?buildId=132002)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/132002/logs/63)|dotnet/runtime#80429|
|[132003](https://dev.azure.com/dnceng-public/public/_build/results?buildId=132003)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/132003/logs/36)|dotnet/runtime#80429|
|[131863](https://dev.azure.com/dnceng-public/public/_build/results?buildId=131863)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/131863/logs/93)|dotnet/runtime#80234|
|[131992](https://dev.azure.com/dnceng-public/public/_build/results?buildId=131992)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/131992/logs/36)|dotnet/runtime#79945|
|[131991](https://dev.azure.com/dnceng-public/public/_build/results?buildId=131991)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/131991/logs/61)|dotnet/runtime#79945|
|[131864](https://dev.azure.com/dnceng-public/public/_build/results?buildId=131864)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/131864/logs/54)|dotnet/runtime#80234|
|[131973](https://dev.azure.com/dnceng-public/public/_build/results?buildId=131973)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/131973/logs/40)|dotnet/runtime#80426|
|[131972](https://dev.azure.com/dnceng-public/public/_build/results?buildId=131972)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/131972/logs/64)|dotnet/runtime#80426|
|[131961](https://dev.azure.com/dnceng-public/public/_build/results?buildId=131961)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/131961/logs/61)|dotnet/runtime#79449|
|[131962](https://dev.azure.com/dnceng-public/public/_build/results?buildId=131962)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/131962/logs/40)|dotnet/runtime#79449|
|[131953](https://dev.azure.com/dnceng-public/public/_build/results?buildId=131953)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/131953/logs/52)|dotnet/runtime#80364|
|[131954](https://dev.azure.com/dnceng-public/public/_build/results?buildId=131954)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/131954/logs/36)|dotnet/runtime#80364|
|[131942](https://dev.azure.com/dnceng-public/public/_build/results?buildId=131942)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/131942/logs/36)|dotnet/runtime#80431|
|[131794](https://dev.azure.com/dnceng-public/public/_build/results?buildId=131794)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/131794/logs/40)|dotnet/runtime#80136|
|[131815](https://dev.azure.com/dnceng-public/public/_build/results?buildId=131815)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/131815/logs/55)|dotnet/runtime#80164|
|[131816](https://dev.azure.com/dnceng-public/public/_build/results?buildId=131816)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/131816/logs/40)|dotnet/runtime#80164|
|[131789](https://dev.azure.com/dnceng-public/public/_build/results?buildId=131789)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/131789/logs/36)|dotnet/runtime#79945|
|[131793](https://dev.azure.com/dnceng-public/public/_build/results?buildId=131793)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/131793/logs/54)|dotnet/runtime#80136|
|[131788](https://dev.azure.com/dnceng-public/public/_build/results?buildId=131788)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/131788/logs/60)|dotnet/runtime#79945|
|[131754](https://dev.azure.com/dnceng-public/public/_build/results?buildId=131754)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/131754/logs/326)|dotnet/runtime#79346|
|[131772](https://dev.azure.com/dnceng-public/public/_build/results?buildId=131772)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/131772/logs/41)|dotnet/runtime#80234|
|[131756](https://dev.azure.com/dnceng-public/public/_build/results?buildId=131756)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/131756/logs/36)|dotnet/runtime#79346|
|[131732](https://dev.azure.com/dnceng-public/public/_build/results?buildId=131732)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/131732/logs/40)|dotnet/runtime#77386|
|[131746](https://dev.azure.com/dnceng-public/public/_build/results?buildId=131746)|dotnet/runtime|Install native dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/131746/logs/403)|dotnet/runtime#80423|
|[131729](https://dev.azure.com/dnceng-public/public/_build/results?buildId=131729)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/131729/logs/815)|dotnet/runtime#77386|
|[131723](https://dev.azure.com/dnceng-public/public/_build/results?buildId=131723)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/131723/logs/53)|dotnet/runtime#80421|
|[131755](https://dev.azure.com/dnceng-public/public/_build/results?buildId=131755)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/131755/logs/62)|dotnet/runtime#79346|
|[131748](https://dev.azure.com/dnceng-public/public/_build/results?buildId=131748)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/131748/logs/294)|dotnet/runtime#76444|
|[131752](https://dev.azure.com/dnceng-public/public/_build/results?buildId=131752)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/131752/logs/36)|dotnet/runtime#76444|
|[131749](https://dev.azure.com/dnceng-public/public/_build/results?buildId=131749)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/131749/logs/37)|dotnet/runtime#80423|
|[131722](https://dev.azure.com/dnceng-public/public/_build/results?buildId=131722)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/131722/logs/527)|dotnet/runtime#80421|
|[131737](https://dev.azure.com/dnceng-public/public/_build/results?buildId=131737)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/131737/logs/43)||
|[131743](https://dev.azure.com/dnceng-public/public/_build/results?buildId=131743)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/131743/logs/41)||
|[131741](https://dev.azure.com/dnceng-public/public/_build/results?buildId=131741)|dotnet/runtime|Install native dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/131741/logs/47)||
|[131731](https://dev.azure.com/dnceng-public/public/_build/results?buildId=131731)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/131731/logs/60)|dotnet/runtime#77386|
|[131704](https://dev.azure.com/dnceng-public/public/_build/results?buildId=131704)|dotnet/runtime|Install native dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/131704/logs/1279)|dotnet/runtime#80260|
|[131705](https://dev.azure.com/dnceng-public/public/_build/results?buildId=131705)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/131705/logs/57)|dotnet/runtime#80260|
|[131666](https://dev.azure.com/dnceng-public/public/_build/results?buildId=131666)|dotnet/runtime|Install native dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/131666/logs/1432)||
|[131648](https://dev.azure.com/dnceng-public/public/_build/results?buildId=131648)|dotnet/runtime|Install native dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/131648/logs/587)|dotnet/runtime#76642|
|[131650](https://dev.azure.com/dnceng-public/public/_build/results?buildId=131650)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/131650/logs/39)|dotnet/runtime#76642|
|[131635](https://dev.azure.com/dnceng-public/public/_build/results?buildId=131635)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/131635/logs/337)|dotnet/runtime#79594|
|[131637](https://dev.azure.com/dnceng-public/public/_build/results?buildId=131637)|dotnet/runtime|Install Build Dependencies|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/131637/logs/36)|dotnet/runtime#79594|
#### Summary
|24-Hour Hit Count|7-Day Hit Count|1-Month Count|
|---|---|---|
|53|53|53|
<!--Known issue error report end -->

This seems to impact only lanes which build for Apple as they use Homebrew. It fails to install OpenSSL: 
```
Installing openssl@3 has failed!
Using pkg-config
Using python3
Homebrew Bundle failed! 1 Brewfile dependency failed to install.
##[error]Bash exited with code '1'.
```

--- a/eng/install-native-dependencies.sh
+++ b/eng/install-native-dependencies.sh
@@ -24,7 +24,10 @@ if [ "$os" = "linux" ] && { [ "$ID" = "debian" ] || [ "$ID_LIKE" = "debian" ]; }
 elif [ "$os" = "maccatalyst" ] || [ "$os" = "osx" ] || [ "$os" = "macos" ] || [ "$os" = "tvos" ] || [ "$os" = "ios" ]; then
     echo "Installed xcode version: $(xcode-select -p)"
 
-    brew update --preinstall
+    export HOMEBREW_NO_INSTALL_CLEANUP=1
+    export HOMEBREW_NO_INSTALLED_DEPENDENTS_CHECK=1
+    # Skip brew update for now, see https://github.com/actions/setup-python/issues/577
+    # brew update --preinstall
     brew bundle --no-upgrade --no-lock --file "$(dirname "$0")/Brewfile"
 else
     echo "Must pass 'Linux', 'macOS', 'maccatalyst', 'iOS' or 'tvOS' as first argument."

